### PR TITLE
recipes-smt: Add recipe for Yices2 SMT solver

### PR DESCRIPTION
**Pull request by ovpanait**
Yices 2 is an SMT solver that decides the satisfiability of formulas
containing uninterpreted function symbols with equality, real and
integer arithmetic, bitvectors, scalar types, and tuples. Yices 2
supports both linear and nonlinear arithmetic. For more details please
see:
https://yices.csl.sri.com/

Yices can be used in conjunction with yosys-smtbmc to formally verify a
design. For more details please see:
https://symbiyosys.readthedocs.io/en/latest/quickstart.html

Signed-off-by: Ovidiu Panait <ovpanait@gmail.com>

### CODE REVIEW

File: conf/layer.conf [hunk 1]
```diff
@@ -1,5 +1,6 @@
 BBPATH .= ":${LAYERDIR}"
 
+BBFILES += "${LAYERDIR}/recipes-smt/*/*.bb"
```
Comment (nathanrossi) [inline conf/layer.conf]: It does not look like there is a very large number of SMT solvers that are supported by yosys-smtbmc. I am thinking it is probably worth creating/locating these in `recipes-support` instead, that way there is a good location in the layer for recipes that are supporting a tool/workflow, but that are themselves not a HDL specific tool/library.

File: recipes-smt/yices2/yices2_git.bb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+PV = "2.6.1+git${SRCPV}"
+
+inherit autotools autotools-brokensep
+
+DEPENDS = " \
+	gperf-native \
+	gmp \
+	"
+
+EXTRA_OEMAKE += "YICES_MAKE_INCLUDE=configs/make.include.${TARGET_SYS} YICES_MODE=debug"
```
Comment (nathanrossi) [inline recipes-smt/yices2/yices2_git.bb]: Is the `YICES_MODE=debug` required? I did not look into what is does specifically, though I suspect you may have added it to resolve `already-stripped` QA warnings/errors.

**Review by nathanrossi — CHANGES_REQUESTED** [review]

File: recipes-smt/yices2/yices2_git.bb (hunk 2 above)
Comment (ovpanait) [reply recipes-smt/yices2/yices2_git.bb]: Yes, YICES_MODE=debug was added in order to resolve the already-stripped QA warnings/errors. By default, YICES_MODE is set to "release" which will strip all the binaries.

The modes are described in https://github.com/SRI-CSL/yices2/blob/master/doc/COMPILING , section "BUILD COMMANDS".

File: conf/layer.conf (hunk 1 above)
Comment (ovpanait) [reply conf/layer.conf]: Good suggestion, will do.

File: recipes-smt/yices2/yices2_git.bb (hunk 2 above)
Comment (nathanrossi) [reply recipes-smt/yices2/yices2_git.bb]: That document states that debug disables optimizations, which I imagine would be useful. Is it possible to trick the makefile to skip stripping similar to how yosys is tricked by swapping swapping `strip` for `echo`? (https://github.com/nathanrossi/meta-hdl/blob/master/recipes-synthesis/yosys/yosys_git.bb#L61)

Comment (ovpanait) [reply recipes-smt/yices2/yices2_git.bb]: Thanks for the suggestion! This is indeed a better approach. I have updated the commit.

**Comment by nathanrossi** [comment]
You accidentally committed a `.swp` file in you most recent commit. So I manually merged this pull request and removed that from the commit. Thanks for the contribution!

**Pull request closed without merging**